File: modules/data/src/main/java/org/gjt/jclasslib/structures/attributes/UninitializedVerificationTypeEntry.java
/*
 This library is free software; you can redistribute it and/or
 modify it under the terms of the GNU General Public
 License as published by the Free Software Foundation; either
 version 2 of the license, or (at your option) any later version.
 */

/*
 This library is free software; you can redistribute it and/or
 modify it under the terms of the GNU General Public
 License as published by the Free Software Foundation; either
 version 2 of the license, or (at your option) any later version.
 */

package org.gjt.jclasslib.structures.attributes;

import org.gjt.jclasslib.structures.InvalidByteCodeException;

import java.io.DataInput;
import java.io.DataOutput;
import java.io.IOException;


/**
 * Describes an entry of type <tt>VerificationType.UNINITIALIZED</tt> in a <tt>BootstrapMethods</tt> attribute structure.
 */
public class UninitializedVerificationTypeEntry extends VerificationTypeInfoEntry {

    public UninitializedVerificationTypeEntry() {
        super(VerificationType.UNINITIALIZED);
    }

    private int offset;

    public int getOffset() {
        return offset;
    }

    public void setOffset(int offset) {
        this.offset = offset;
    }

    @Override
    protected void readExtra(DataInput in) throws InvalidByteCodeException, IOException {
        super.readExtra(in);
        offset = in.readUnsignedShort();
    }

    @Override
    public void writeExtra(DataOutput out) throws InvalidByteCodeException, IOException {
        super.writeExtra(out);
        out.writeShort(offset);
    }

    @Override
    public void appendTo(StringBuilder buffer) {
        super.appendTo(buffer);
        buffer.append(" (offset: ").append(offset).append(")");
    }

    @Override
    public int getLength() {
        return super.getLength() + 2;
    }
}

File: modules/data/src/main/java/org/gjt/jclasslib/structures/attributes/UninitializedVerificationTypeEntry.kt
/*
 This library is free software; you can redistribute it and/or
 modify it under the terms of the GNU General Public
 License as published by the Free Software Foundation; either
 version 2 of the license, or (at your option) any later version.
 */

/*
 This library is free software; you can redistribute it and/or
 modify it under the terms of the GNU General Public
 License as published by the Free Software Foundation; either
 version 2 of the license, or (at your option) any later version.
 */

package org.gjt.jclasslib.structures.attributes

import java.io.DataInput
import java.io.DataOutput


/**
 * Describes an entry of type VerificationType.UNINITIALIZED in a StackMapFrameEntry attribute structure.
 */
class UninitializedVerificationTypeEntry : VerificationTypeInfoEntry(VerificationType.UNINITIALIZED) {

    var offset: Int = 0

    override fun read(input: DataInput) {
        super.read(input)
        offset = input.readUnsignedShort()
        if (isDebug) debug("read")
    }

    override fun write(output: DataOutput) {
        super.write(output)
        output.writeShort(offset)
        if (isDebug) debug("wrote")
    }

    override fun appendTo(buffer: StringBuilder) {
        super.appendTo(buffer)
        buffer.append(" (offset: ").append(offset).append(")")
    }

    override val length: Int
        get() = super.length + 2
}
